crates/noseyparker/data/default/builtin/rules/truenas.yml

```diff
@@ -0,0 +1,21 @@
+rules:
+
+- name: TrueNAS API Key
+  id: np.truenas.1
+
+  pattern: |
+    (?x)
+    \b
+    (\d+-[a-zA-Z0-9]{64})
+    \b
```

Comment:
I read some documentation and looked at other examples online, and it does appear that this matches TrueNAS API keys.

However, this pattern lacks a distinctive prefix or suffix, and is going to match many other things that are not TrueNAS API Keys, such as:
- filenames: `"fonts/fontawesome-webfont-v=4.5.0-7b5a4320fba0d4c8f79327645b4b9cc875a2ec617a557e849b813918eb733499.ttf"`, `"products/signle-product/product-slide-small-01-2bc7fc9f7b3c9f90fe5d98b9884f84addba83e33781756d9aae669d50fd5bc2d.jpg"`, or `"aEwP539_460sa-1-30504e59566e451558d63cc97209d6a111867d65257ab650ecc577d6d9b6217c.gif"`
- fstab entries, such as `/ /var/lib/docker/devicemapper/mnt/7770c8b24eb3d5cc159a065910076938910d307ab2f5d94e1dc3b24c06ee2c8a rw,relatime shared:205 - ext4 /dev/mapper/docker-253:2-425882-7770c8b24eb3d5cc159a065910076938910d307ab2f5d94e1dc3b24c06ee2c8a rw,seclabel,discard,stripe=16,data=ordered`

To keep the rule from being noisy, is there some additional piece of context that could be added to the rule? (This is a balancing act between making the rule more precise and keeping it simple / avoiding overfitting.) For example, perhaps something like this:
```suggestion
  pattern: |
    (?x)(?i)
    (?: nas | key | bearer ) .{0,20}
    \b
    (\d+-[a-zA-Z0-9]{64})
    \b
```

Reply:
I did something similar in the end, check out the new patterns.